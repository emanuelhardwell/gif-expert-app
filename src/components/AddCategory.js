import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
    console.log("handleInputChange ...");
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    // Se hace de esta forma ya que unicamente estamos pasando la FUNCION setCategories
    // y usamos un CALLBACK para tener la referencia de de las categorias en este caso
    // o bien podriamos pasar por las PROPS a CATEGORIES y quedaria asi ------ >
    //setCategories([...categories, "HunterXHunter"]);

    console.log("handleSubmit ...");
    console.log("---- " + inputValue + " ----");

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };

  // RETURN
  return (
    <>
      <form onSubmit={handleSubmit}>
        <p>{inputValue}</p>
        <input
          className="form-control"
          type="text"
          name="buscar"
          placeholder="Buscar por categoria"
          value={inputValue}
          onChange={handleInputChange}
        />
      </form>
    </>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
