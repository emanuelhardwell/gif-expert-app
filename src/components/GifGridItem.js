import React from "react";
import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
  return (
    <>
      <div className="card">
        <img src={url} className="card-img-top" alt={title}></img>
        <div className="card-body">
          <p className="card-title"> {title} </p>
        </div>
      </div>
    </>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
