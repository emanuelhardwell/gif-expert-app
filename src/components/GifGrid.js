import React, { useEffect, useState } from "react";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?q=goku&limit=8&api_key=EMkLUTqDMhIksWGffC42PtHNNshPjgwj";
    const resp = await fetch(url);
    const { data } = await resp.json();
    // console.log(data);

    const gifs = data.map((image) => {
      return {
        id: image.id,
        title: image.title,
        url: image.images?.downsized?.url,
      };
    });

    console.log(gifs);
    setImages(gifs);
  };

  // RETURN
  return (
    <>
      <h3 className="text-center"> {category} </h3>
      <div className="container">
        <div className="row bg-dark">
          <div className="col-md-4 mb-3">
            {images.map((img) => (
              <GifGridItem key={img.id} {...img} />
            ))}
          </div>
        </div>
      </div>
    </>
  );
};
