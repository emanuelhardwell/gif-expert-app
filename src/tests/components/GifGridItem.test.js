// Debe mostrar el componente correctamente
//shalow
// wrapper
//wrapper.toMatchSnapshot

import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Prueba en <GifGridItem />", () => {
  const title = "title de la imagen";
  const url = "http://localhost/url-de-la-imagen.jpg";

  test("should mostrar el componente correctamente", () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
  });
});
