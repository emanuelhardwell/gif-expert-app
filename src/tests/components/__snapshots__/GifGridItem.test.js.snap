// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem /> should mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="card"
  >
    <img
      alt="title de la imagen"
      className="card-img-top"
      src="http://localhost/url-de-la-imagen.jpg"
    />
    <div
      className="card-body"
    >
      <p
        className="card-title"
      >
         
        title de la imagen
         
      </p>
    </div>
  </div>
</Fragment>
`;
